### PR TITLE
Add Spring Session

### LINKED ISSUE #1402: Add user session
### Description
As part of the custom create account registration the backend will need to recall information about the current user between form submissions. The first step in the flow will authenticate a user with an okta account via an authorization token passed in from the frontend. from that point forward the users session will be used to identify the okta user, and MFA information like the factorID, SMS resend link, etc.

### Proposed solution 
- Add a database backed session [spring-session-jdbc](https://www.baeldung.com/spring-session-jdbc)

### PR DESCRIPTION
**Pull request by emmastephenson**
## Related Issue or Background Info

Fixes #1402 

## Changes Proposed

- Adds SpringSession to the app, backed by our regular postgres database.

## Additional Information

- followup PR that uses SpringSession is [PR/1443](https://github.com/CDCgov/prime-simplereport/pull/1443)

### UI
- ~~[ ] Any changes to the UI/UX have been approved by design~~
- ~~[ ] Any new or edited error messages have been approved by design~~

### Testing
- ~~[ ] Includes a summary of what a code reviewer should verify~~

### Changes are Backwards Compatible
- [x] Database changes are submitted as a separate PR
- [x] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

## Cloud
- [ ] DevOps team has been notified if PR requires ops support

### CODE REVIEW

File: backend/src/main/resources/application.yaml [hunk 1]
```diff
@@ -19,6 +19,8 @@ spring:
   jackson:
     serialization:
       FAIL_ON_EMPTY_BEANS: false
+  session:
+    store-type: jdbc
```
Comment (jeskew-gov) [inline backend/src/main/resources/application.yaml:23]: The session will need to time out after 15 minutes. From [the Spring docs](https://docs.spring.io/spring-boot/docs/current/reference/html/appendix-application-properties.html#spring.session.timeout), I think this is the correct syntax:

```suggestion
  session:
    store-type: jdbc
    timeout: 900
```

**Comment by jeskew-gov** [comment]
I think as part of this PR, we will need to re-enable CSRF (as [this comment](https://github.com/CDCgov/prime-simplereport/blob/78e1d2390b3ebdbd9a0accc3d161f071fc3a3e58/backend/src/main/java/gov/cdc/usds/simplereport/config/SecurityConfiguration.java#L80-L84) would no longer hold true). Beyond removing those lines from configuration, I think that would also require including CSRF tokens in the existing patient experience forms.

File: backend/src/main/resources/application.yaml (hunk 1 above)
Comment (emmastephenson) [reply backend/src/main/resources/application.yaml:23]: Good suggestion, thank you! Updated.

**Comment by ClydeDroid** [comment]
> I think as part of this PR, we will need to re-enable CSRF (as [this comment](https://github.com/CDCgov/prime-simplereport/blob/78e1d2390b3ebdbd9a0accc3d161f071fc3a3e58/backend/src/main/java/gov/cdc/usds/simplereport/config/SecurityConfiguration.java#L80-L84) would no longer hold true). Beyond removing those lines from configuration, I think that would also require including CSRF tokens in the existing patient experience forms.

I think this would require frontend changes as well - the tricky thing about the patient experience though is that there is no authentication first, so the request to the backend won't have a session yet and therefore no CSRF token to pass along. @nullflux you had done some research into this IIRC - is it possible to do CSRF with the patient experience?

**Comment by ClydeDroid** [comment]
Normally you would set a cookie in the browser[ like this](https://stackoverflow.com/questions/54345301/react-springboot-csrf). I suppose we could disable CSRF just for the first verification request, and then set the cookie after the session is established?

**Comment by jeskew-gov** [comment]
> I think this would require frontend changes as well - the tricky thing about the patient experience though is that there is no authentication first, so the request to the backend won't have a session yet and therefore no CSRF token to pass along.

@emmastephenson and I chatted about this in zoom. The patient experience operations on the backend all require authentication on every request, so CSRF isn't needed there. (In theory, this could change; the patient could verify themselves and create a session, then the ask-on-entry form could require a session ID and a CSRF token, but that's not how it works today.)

We will need to selectively re-enable Spring's CSRF protection when the self-registration flow requires the user to follow a particular sequence of actions, but that's not necessary yet.

**Review by jeskew-gov — APPROVED** [review]

**Review by ClydeDroid — APPROVED** [review]

**Pull request merged**